pypif_sdk/func/calculate_funcs.py

```diff
@@ -0,0 +1,258 @@
+import pypif
+import re
+from mendeleev import element
+from pypif.obj import *
+
+def calculate_ideal_atomic_percent(pif):
+    """
+    Calculates ideal atomic percents from a chemical formula string from a pif. Returns an appended pif with composition elements modified or added.
+    """
+    try:
+        pif.chemicalFormula
+    except NameError:
+        return pif
```

Comment:
When does this happen?  If its just when the pif is a System but not ChemicalSystem, could we just check if its a ChemicalSystem rather than testing an exception?

Reply:
There can be no chemical formula even in a ChemicalSystem if it wasn't set, also. In which case my functions are worthless.

Reply:
Wouldn't it be `None` if it wasn't set?